File: server/src/validation/exam.ts
import { z } from "zod";

export const createExamSchema = z.object({
  title: z.string().min(1, "Exam title is required"),
  subject: z.string().min(1, "Subject is required"),
  date: z.string().regex(/^\d{4}-\d{2}-\d{2}$/, "Date must be in format YYYY-MM-DD"),
  time: z.string().regex(/^([0-9]|0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]$/, "Invalid time format").optional().nullable(),
  notes: z.string().optional().nullable(),
});

File: backend/src/main/java/com/timeforge/model/Exam.java
package com.timeforge.model;

import jakarta.persistence.*;
import java.util.UUID;

@Entity
@Table(name = "exams")
public class Exam {
    @Id
    private String id = UUID.randomUUID().toString();

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String subject;

    @Column(nullable = false)
    private String date;

    private String time;
    private String notes;

    @Column(nullable = false)
    private String userId;

    public Exam() {}

    @PrePersist
    protected void onCreate() {
        if (id == null) id = UUID.randomUUID().toString();
    }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public String getSubject() { return subject; }
    public void setSubject(String subject) { this.subject = subject; }
    public String getDate() { return date; }
    public void setDate(String date) { this.date = date; }
    public String getTime() { return time; }
    public void setTime(String time) { this.time = time; }
    public String getNotes() { return notes; }
    public void setNotes(String notes) { this.notes = notes; }
    public String getUserId() { return userId; }
    public void setUserId(String userId) { this.userId = userId; }
}
